cli/src/main/java/com/box/l10n/mojito/cli/command/ExtractionDiffCommand.java

```diff
@@ -251,8 +282,33 @@ protected Optional<SlackNotificationSender> getNotificationSender() {
     return empty();
   }
 
+  private void validateBranchDeletionParams() {
+    if (this.githubRepository != null && this.githubPrNumber == null) {
+      throw new CommandException(
+          "Github PR number must be specified when Github Repository is provided");
+    }
+    if (this.githubPrNumber != null && this.githubRepository == null) {
+      throw new CommandException(
+          "Github Repository must be specified when Github PR number is provided");
+    }
+    if (this.githubRepository != null && this.pushToBranchName == null) {
+      throw new CommandException(
+          "The Push To Branch parameter must be specified when Github Repository is provided");
+    }
+    if (this.githubRepository != null && this.owner == null) {
+      throw new CommandException(
+          "The Github owner must be specified when Github Repository is provided");
+    }
+  }
+
   @Override
   public void execute() throws CommandException {
+    if (this.owner != null && this.githubClients == null) {
+      throw new CommandException(
+          "Github must be configured with properties: l10n.githubClients.<client>.appId, l10n.githubClients.<client>.key and l10n.githubClients.<client>.owner");
+    }
+    this.validateBranchDeletionParams();
```

Comment:
If no Github parameters are provided we should disable the branch deletion logic rather than failing the command. Not all users will be using Github as their VCS